cdap-common/bin/functions.sh

```diff
@@ -0,0 +1,1085 @@
+      ${__command} ${__foreground} ${__debug} ${__port} ${__arg}
+      __ret=${?}
+      ;;
+    status) cdap_status_pidfile ${__pidfile} "CDAP Standalone (SDK)"; __ret=${?} ;;
+    stop) cdap_sdk_stop; __ret=${?} ;;
+    usage) cdap_sdk_usage; __ret=${?} ;;
+    cleanup) cdap_sdk_cleanup; __ret=${?} ;;
+    *) cdap_sdk_usage; __ret=1 ;; # Return non-zero; called incorrectly
+  esac
+  return ${__ret}
+}
+
+# cdap_tx_debugger
+cdap_tx_debugger() {
+  local readonly __path __libexec __lib __script="$(basename ${0}):cdap_tx_debugger"
+  local readonly __authfile="${HOME}"/.cdap.accesstoken
+  local readonly __ret __class=co.cask.cdap.data2.transaction.TransactionManagerDebuggerMain
+  cdap_set_java || die "Unable to locate JAVA or JAVA_HOME"
+  __path=$(cdap_home)
+  if [[ -d ${__path}/master/libexec ]]; then
+    __libexec=${__path}/master/libexec
+    __lib=${__path}/master/lib
+  else
+    __libexec=${__path}/libexec
+    __lib=${__path}/lib
+  fi
+  if [[ ${CLASSPATH} == "" ]]; then
+    CLASSPATH=${__lib}/*
+  else
+    CLASSPATH=${CLASSPATH}:${__lib}/*
+  fi
+  if [[ -d ${CDAP_CONF} ]]; then
+    CLASSPATH=${CLASSPATH}:"${CDAP_CONF}"
+  elif [[ -d ${__path}/conf ]]; then
+    CLASSPATH=${CLASSPATH}:"${__path}"/conf/
+  fi
+  # add token file arg with default token file if one is not provided
+  local __has_arg=0 __var
+  for __var in ${@}; do
+    if [[ ${__var} == "--token-file" ]]; then
+      __has_arg=1
+    fi
+  done
+  if [[ ${__has_arg} -eq 0 ]] && [[ -f ${__auth_file} ]]; then
+    set -- ${@} "--token-file" "${__auth_file}"
+  fi
+
+  "${JAVA}" -cp ${CLASSPATH} -Dscript=${__script} ${__class} ${@}
+  __ret=${?}
+  return ${__ret}
+}
+
+#
+# User-definable variables
+CDAP_NODE_VERSION_MINIMUM=${CDAP_NODE_VERSION_MINIMUM:-v0.10.36}
+
+# Specifies CDAP UI Path
+CDAP_UI_PATH=${CDAP_UI_PATH:-ui/server.js}
+
+# Default CDAP_CONF to /etc/cdap/conf (package default)
+export CDAP_CONF=${CDAP_CONF:-/etc/cdap/conf}
+
+# Set CDAP_HOME
+export CDAP_HOME=$(cdap_home)
+
+# Make sure HOSTNAME is in the environment
+export HOSTNAME=$(hostname -f)
+
+# The directory serving as the user directory for master
+export LOCAL_DIR=${CDAP_LOCAL_DIR:-/var/tmp/cdap}
+
+# Where log files are stored.
+export LOG_DIR=${CDAP_LOG_DIR:-/var/log/cdap}
+
+# A string representing this instance of CDAP. $USER by default.
+export IDENT_STRING=${USER}
+
+# Extra Java runtime options.
+# Below are what we set by default.  May only work with SUN JVM.
+# For more on why as well as other possible settings,
+# see http://wiki.apache.org/hadoop/PerformanceTuning
+export OPTS="-XX:+UseConcMarkSweepGC -XX:MaxPermSize=256m"
+
+# The directory where PID files are stored. Default: /var/cdap/run
+export PID_DIR=${CDAP_PID_DIR:-/var/cdap/run}
+
+# The directory serving as the java.io.tmpdir directory for master
+export TEMP_DIR=${CDAP_TEMP_DIR:-/tmp}
+
+# Add default JVM options here.
+CDAP_OPTS="${OPTS} -Djava.security.krb5.realm= -Djava.security.krb5.kdc= -Djava.awt.headless=true"
```

Comment:
these are only used in the ``cdap_sdk_start``... maybe we should rename to ``CDAP_SDK_OPTS`` or similar.  Otherwise its misleading (with comment above as well)